### PR TITLE
DT-2728: Migrate DAR.js to Typescript

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
https://broadworkbench.atlassian.net/browse/DT-2728

## Summary
Migrate `DAR.js` to Typescript 

### Reviewer Note
I used `.claude/commands/convert-ajax.md` for the migration and addressed a number of sonar issues individually. I'm open to different opinions on keeping it as part of the codebase. I intend to reuse it for the other ajax lib conversions as well.

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/libs/ajax/DAR.js
Comment (rushtong) [inline src/libs/ajax/DAR.js:1]: Note about GH not respecting `git mv`: even with a single commit, GitHub's rename detection may still miss it if the TypeScript rewrite adds enough lines to push similarity below ~50% — JSDoc comments are a particular culprit on smaller files.

Comment (rushtong) [inline src/libs/ajax/DAR.js:1]: Note about GH not respecting `git mv`: even with a single commit, GitHub's rename detection may still miss it if the TypeScript rewrite adds enough lines to push similarity below ~50% — JSDoc comments are a particular culprit on smaller files.

File: src/libs/ajax/DAR.ts [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+  /**
+   * Create a new draft DAR.
+   * @param dar The initial DAR form data
+   * @returns The newly created DataAccessRequest
+   */
+  postDarDraft: async (dar: Record<string, unknown>): Promise<DataAccessRequest & { code?: number, message?: string }> => {
+    // noinspection ES6MissingAwait
+    Metrics.captureEvent(eventList.dar, { action: 'draft' })
+    const url = `${await Config.getApiUrl()}/api/dar/v2/draft`
+    const res = await fetchPost<DataAccessRequest & { code?: number, message?: string }>(url, dar, Config.authOpts())
```
Comment (kevinmarete) [inline src/libs/ajax/DAR.ts]: Is including `& { code?: number, message?: string }` necessary?

File: src/libs/ajax/DAR.ts [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+  /**
+   * Approve the closeout of a DAR.
+   * @param referenceId The DAR reference ID
+   * @returns The HTTP status from the response envelope (undefined in practice; preserved for API compatibility)
+   */
+  approveCloseout: async (referenceId: string): Promise<number | undefined> => {
+    const url = `${await Config.getApiUrl()}/api/dar/${referenceId}/approveCloseout`
+    const res = await fetchPut(url, {}, Config.authOpts())
+    // FetchData only exposes .data; .status is always undefined here, preserved for API compatibility
+    return (res as unknown as { status?: number }).status
```
Comment (kevinmarete) [inline src/libs/ajax/DAR.ts]: This can be simplified to `return 200` then the caller can catch error codes and messages.

**Review by kevinmarete — APPROVED** [review]
Left some comments, but looks good to me!

File: src/libs/ajax/DAR.ts (hunk 1 above)
Comment (rushtong) [reply src/libs/ajax/DAR.ts]: Good point, updating

Comment (rushtong) [reply src/libs/ajax/DAR.ts]: There was a caller that was referencing message, but ended up being dead code.

File: .claude/commands/convert-ajax.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+   - For every method that propagates fetch errors (i.e. does NOT catch them internally), add a ConsentError test: reject the fetch mock with `{ message: '<descriptive message>', code: <status> }` (the `ConsentError` shape from `src/types/model.ts`), capture the rejection, and assert that the real (unmocked) `extractConsentError` and `extractError` from `src/utils/ErrorUtils` recover the shape and message. Always expand the `.then(onFulfilled, onRejected)` call across multiple lines — putting `() => { throw ... }` inline triggers the `@stylistic/max-statements-per-line` lint rule. The one exception is methods that catch all errors internally and return a fallback (e.g. `searchOntologyIdList` returns `[]` on error) — those swallow ConsentErrors too, so no test is needed. See `test/libs/ajax/DAR.spec.ts` for examples of this pattern across multiple methods.
+   - Commit as `test: add tests`.
+5. Verify, fixing anything that fails before finishing:
+   - `npm run type-check`
+   - `npm run test -- test/libs/ajax/<Name>.spec.ts`
+   - `npx eslint src/libs/ajax/<Name>.ts test/libs/ajax/<Name>.spec.ts`
```
Comment (fboulnois) [inline .claude/commands/convert-ajax.md]:
```suggestion
   - `pnpm run type-check`
   - `pnpm run test -- test/libs/ajax/<Name>.spec.ts`
   - `pnpx eslint src/libs/ajax/<Name>.ts test/libs/ajax/<Name>.spec.ts`
```

**Review by fboulnois — APPROVED** [review]
One suggestion:

File: .claude/commands/convert-ajax.md (hunk 3 above)
Comment (rushtong) [reply .claude/commands/convert-ajax.md]: Great catch, thank you!
